bindings/c/test/mako/mako.cpp

```diff
@@ -136,29 +153,38 @@ int cleanup(Database db, Arguments const& args) {
 	} else {
 		int batch_size = args.tenant_batch_size;
 		int batches = (args.total_tenants + batch_size - 1) / batch_size;
-		fdb::TypedFuture<fdb::future_var::ValueRef> tenantResults[batch_size];
-		// Issue all tenant reads first
-		Transaction getTx = db.createTransaction();
-		for (int i = 0; i < args.total_tenants; ++i) {
-			std::string tenant_name = "tenant" + std::to_string(i);
-			tenantResults[i] = Tenant::getTenant(getTx, toBytesRef(tenant_name));
-		}
 		// First loop to clear all tenant key ranges
 		for (int batch = 0; batch < batches; ++batch) {
+			fdb::TypedFuture<fdb::future_var::ValueRef> tenantResults[batch_size];
+			// Issue all tenant reads first
+			Transaction getTx = db.createTransaction();
+			for (int i = batch * batch_size; i < args.total_tenants && i < (batch + 1) * batch_size; ++i) {
+				std::string tenant_name = "tenant" + std::to_string(i);
+				tenantResults[i - (batch * batch_size)] = Tenant::getTenant(getTx, toBytesRef(tenant_name));
```

Comment:
I think this is fine for now, but we should plan to switch to reading the tenants as a list to try to get this runtime down later.